hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsScheduler.java

```diff
@@ -127,6 +126,10 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
       status.getMessage(),
       liveState.getStagingTasksSize()));
 
```

Comment:
The scheduler should be able to modify state. I wouldn't consider that a sin. It is making calls to update the state. Then this functionality is actually executed by the state stores. To me, the worst sin of the Scheduler is that there is too much logic there. We need to spend some time removing this tech debt and breaking down the functionality into smaller pieces.

Regarding observers, I don't want to unnecessarily make any calls that aren't required. If the observers ignore these unnecessary calls and don't execute additional logic (other than logging), that is okay with me. As you mentioned these are only a few calls so this is not a deal breaker. The main point here is to be very careful about what logic we want to execute and not make unnecessary calls.